src/test/compile-fail/divide-by-zero.rs

```diff
@@ -0,0 +1,8 @@
+// Copied from `test/run-fail`.
```

Comment:
You can move this test under `src/test/ui`

Reply:
I did! I put them in `compile-fail` and in `ui/mir`. Here we check just that they always fail to compile. There we check how the ui should look.

Reply:
The `compile-fail` tests are being migrated into ui tests. We shouldn't be adding more `compile-fail` tests unless we have to. (UI tests can check that code fails to compile as well. You just need to include `//~ ERROR` annotations)

Reply:
Ahh! Excellent.